core/src/main/java/org/jclouds/rest/internal/RestAnnotationProcessor.java

```diff
@@ -763,6 +763,8 @@ private static void addHeader(Multimap<String, String> headers, Headers header,
       for (int i = 0; i < header.keys().length; i++) {
          String value = header.values()[i];
          value = replaceTokens(value, tokenValues);
+         if (header.urlEncode().length > i && header.urlEncode()[i])
```

Comment:
[minor] Change to `i < header.urlEncode().length && header.urlEncode()[i]`? The bounds check seems to be written that way around more commonly, which might make it easier to understand that's going on here